lib/puppet-lint/data.rb

```diff
@@ -259,38 +270,38 @@ def node_indexes
     def definition_indexes(type)
       result = []
       tokens.each_with_index do |token, i|
-        if token.type == type
-          brace_depth = 0
-          paren_depth = 0
-          in_params = false
-          inherited_class = nil
-          tokens[i+1..-1].each_with_index do |definition_token, j|
-            case definition_token.type
-            when :INHERITS
-              inherited_class = definition_token.next_code_token
-            when :LPAREN
-              in_params = true if paren_depth == 0
-              paren_depth += 1
-            when :RPAREN
-              in_params = false if paren_depth == 1
-              paren_depth -= 1
-            when :LBRACE
-              brace_depth += 1
-            when :RBRACE
-              brace_depth -= 1
-              if brace_depth == 0 && !in_params
-                if token.next_code_token.type != :LBRACE
-                  result << {
-                    :start           => i,
-                    :end             => i + j + 1,
-                    :tokens          => tokens[i..(i + j + 1)],
-                    :param_tokens    => param_tokens(tokens[i..(i + j + 1)]),
-                    :type            => type,
-                    :name_token      => token.next_code_token,
-                    :inherited_token => inherited_class,
-                  }
-                  break
-                end
+        next unless token.type == type
+
+        brace_depth = 0
+        paren_depth = 0
+        in_params = false
+        inherited_class = nil
+        tokens[i + 1..-1].each_with_index do |definition_token, j|
+          case definition_token.type
+          when :INHERITS
+            inherited_class = definition_token.next_code_token
+          when :LPAREN
+            in_params = true if paren_depth.zero?
+            paren_depth += 1
+          when :RPAREN
+            in_params = false if paren_depth == 1
```

Comment:
This whole method needs to be rewritten (it can be simplified a lot with `PuppetLint::Lexer::Token#next_token_of), but I've deliberately avoided making any logic changes where possible in this PR (just shuffling some loop conditionals around to keep rubocop happy).

My overall plan is:
 1. Clean up code style (this PR)
 2. Audit tests to ensure we have adequate coverage (methods like `PuppetLint::Data.function_indexes` for example currently have no coverage at all)
 3. Refactor to clean up and modernise the PuppetLint core (addressing issues like this)